src/app/GitUI/CommandsDialogs/SettingsDialog/SettingsPageWithHeader.cs

```diff
@@ -1,18 +1,22 @@
 ﻿using GitCommands;
+using GitExtensions.Extensibility.Git;
 using GitExtensions.Extensibility.Settings;
+using GitUIPluginInterfaces;
 
 namespace GitUI.CommandsDialogs.SettingsDialog
 {
     public partial class SettingsPageWithHeader : SettingsPageBase, IGlobalSettingsPage
     {
         private SettingsPageHeader? _header;
+        private bool _canSaveInsideRepo;
 
-        public SettingsPageWithHeader(IServiceProvider serviceProvider)
-            : base(serviceProvider)
+        public SettingsPageWithHeader(IServiceProvider serviceProvider, ISettingsPageHost pageHost)
+            : base(serviceProvider, pageHost)
         {
+            _canSaveInsideRepo = pageHost.CheckSettingsLogic.CommonLogic.Module.IsValidGitWorkingDir();
```

Comment:
What's the concrete type of `serviceProvider`? It's an instance of `GitUICommands`, which implements `IGitUICommands` and, thus, inherits from `IServiceProvider`.
![image](https://github.com/user-attachments/assets/5bb2070f-d974-4ab2-abcf-71436274ff62)

Which makes this possible:
```cs
if (serviceProvider is IGitUICommands uiCommands)
{
    _canSaveInsideRepo = uiCommands.Module.IsValidGitWorkingDir();
}
```

It's a little hacky, but we some point I expect we'll have something like the following:
```cs
IGitUICommands uiCommands = serviceProvider.GetREquiredService<IGitUICommands>();
```

Reply:
> Which makes this possible:

I saw that but it was (too) hacky for me that's why I have not decided to go that way.
It's you to decide...

Reply:
In the end, `ISettingsPageHost`, that still need to be passed is now passed in the constructor to have it earlier than in a later call to do some initialization. It's indeed a lot of code changes but nothing silly...

Reply:
The current change introduces a hard coupling, and this I'd like to avoid. 
Let's resolve from the service provider

Reply:
Done.